frontend-ssr/components/notification-tooltip/NotificationTooltip.component.tsx

```diff
@@ -0,0 +1,234 @@
+      dispatch(loadNotificationsSuccess(newNotifications));
+      // this.scrollRef.current.refresh();
+    } catch (err) {
+      SweetAlertService.toast({ type: 'error', text: err });
+    } finally {
+      this.setState({ loading: false });
+    }
+  };
+
+  markAsRead = async (id) => {
+    const { user } = this.props;
+    const { notifications } = user;
+    const { dispatch } = this.props;
+
+    if (notifications.list.find((n) => n._id === id).isUnread) {
+      try {
+        await UserService.markAsRead(id);
+        dispatch(markNotificationAsRead(id));
+      } catch (err) {
+        dispatch(markNotificationAsUnread(id));
+      }
+    }
+  };
+
+  markAllAsRead = async () => {
+    const { dispatch } = this.props;
+    try {
+      await UserService.markAllAsRead();
+      dispatch(markAllAsRead());
+    } catch (err) {
+      SweetAlertService.toast({ type: 'error', text: err });
+    }
+  };
+
+  render() {
+    const { user } = this.props;
+    const { loading, isOpen } = this.state;
+
+    const notifications = user.notifications.list || [];
+
+    return (
+      <div
+        /** the initial code was { display: !notifications.length ? 'none' : 'initial' }
+       but I kept it this way for now just to see the wrapper */
+        style={{ display: !notifications.length ? '' : 'initial' }}
+        className={`${styles['header-tooltip']} ${styles['notifications-tooltip']}`}
+      >
+        <div className={styles.icon} ref={this.toggleRef}>
+          <i className={`${loading ? 'icon-spinner' : 'icon-bell'}`} />
+          {user.notifications.unreadCount > 0 && (
+            <span className={styles.badge}>
+              {user.notifications.unreadCount}
+            </span>
+          )}
+        </div>
+        {isOpen && (
+          <Scroll
+            hasInfiniteScroll={!user.notifications.end}
+            onEndReach={this.loadNextPage}
+            preloadDistance={PRELOAD_DISTANCE}
+            ref={this.scrollRef}
+          >
+            <ul className={styles.notifications}>
+              <li>
+                <button
+                  type="button"
+                  id={styles.markAllAsReadBtn}
+                  className="outline-none"
+                  onClick={this.markAllAsRead}
+                >
+                  Mark all as read
+                </button>
+              </li>
+              {notifications.map(
+                ({
+                  _id,
+                  username,
+                  name,
+                  avatar,
+                  message,
+                  timestamp,
+                  url,
+                  isUnread,
+                }) => (
+                  <li
+                    key={_id}
+                    className={isUnread ? `${styles['is--unread']}` : ''}
+                    onClick={() => this.markAsRead(_id)}
+                    role="presentation"
+                  >
+                    <Notification
```

Comment:
Let's simplify here. Instead of passing each prop one by one we can take advantage of destructuring:

```
{notifications.map(notificationProps => (
  <li>
     <Notification {...notificationProps} />
  </li>
))}
```